R/data.table.R

```diff
@@ -1123,7 +1123,7 @@ replace_dot_alias = function(e) {
               if (is.list(k)) {
                 origj = j = if (name[[1L]] == "$") as.character(name[[3L]]) else eval(name[[3L]], parent.frame(), parent.frame())
                 if (is.character(j)) {
-                  if (length(j)!=1L) stop("Cannot assign to an under-allocated recursively indexed list -- L[[i]][,:=] syntax is only valid when i is length 1, but it's length ", length(j))
+                  if (length(j)!=1L) stop("Cannot assign to an under-allocated recursively indexed list -- L[[i]][,:=] syntax is only valid when i is length 1, but its length ", length(j))
```

Comment:
it was more right before with "it's" as a contraction of "it is"? But it could be better, agree. Suggest any of the following:
, but it is length 
, but i is length
, but its length is
, but i's length is

Reply:
Fair point! I also noticed we were using "when i is" when it should be "when j is". Updated now. to "its length is" and used a formatted message while we're at it.